OutOfSchool/OutOfSchool.WebApi.Tests/Services/CityServiceTests.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Localization;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.Services;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Models;
+using OutOfSchool.WebApi.Services;
+using Serilog;
+
+namespace OutOfSchool.WebApi.Tests.Services
+{
+    [TestFixture]
+    public class CityServiceTests
+    {
+        private ICityService service;
+        private OutOfSchoolDbContext context;
+        private IEntityRepository<City> repository;
+        private Mock<IStringLocalizer<SharedResource>> localizer;
+        private Mock<ILogger> logger;
+        private DbContextOptions<OutOfSchoolDbContext> options;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var builder = new DbContextOptionsBuilder<OutOfSchoolDbContext>()
+                .UseInMemoryDatabase(databaseName: "OutOfSchoolTestDB");
+
+            options = builder.Options;
+            context = new OutOfSchoolDbContext(options);
+            localizer = new Mock<IStringLocalizer<SharedResource>>();
+            repository = new EntityRepository<City>(context);
+            logger = new Mock<ILogger>();
+            service = new CityService(repository, logger.Object, localizer.Object);
+
+            SeedDatabase();
+        }
+
+        [Test]
+        public async Task GetAll_WhenCalled_ReturnsAllCities()
+        {
+            // Arrange
+            var expected = await repository.GetAll();
+
+            // Act
+            var result = await service.GetAll().ConfigureAwait(false);
+
+            // Assert
+            Assert.AreEqual(result.ToList().Count(), expected.Count());
+        }
+
+        [Test]
+        [TestCase(1)]
+        public async Task GetById_WhenIdIsValid_ReturnsCity(long id)
+        {
+            // Arrange
+            var expected = await repository.GetById(id);
+
+            // Act
+            var result = await service.GetById(id).ConfigureAwait(false);
+
+            // Assert
+            Assert.AreEqual(expected.Id, result.Id);
+        }
+
+        [Test]
+        [TestCase(10)]
+        public void GetById_WhenIdIsInvalid_ThrowsArgumentOutOfRangeException(long id)
+        {
+            // Act and Assert
+            Assert.ThrowsAsync<ArgumentOutOfRangeException>(
+                async () => await service.GetById(id).ConfigureAwait(false));
+        }
+
+        [Test]
+        [TestCase("NoName")]
+        public async Task GetByCityName_WhenNameIsValid_ReturnsCities(string name)
+        {
+            // Arrange
+            var expected = await repository.GetByFilter(c => c.Name.Contains(name));
+
+            // Act
+            var result = await service.GetByCityName(name).ConfigureAwait(false);
+
+            // Assert
+            Assert.AreEqual(result.ToList().Count(), expected.Count());
```

Comment:
can be result.Count()

Reply:
Dear @yfedo, 
Thank you for your advice.
This was added to the project.
Thank you.